### ISSUE TITLE
Authentication failed

### ISSUE THREAD

**Issue by phatcher** [AzureADQuickStarts/B2C-NativeClient-DotNet#1]
The user dialog displays for sign in and I enter my password and click ok.

I always get an error of "authentication_ui_failed" with the message

The browser based authentication dialog failed to complete. Reason: The protocol is not known and no pluggable protocols have been entered that match.

**Comment by dstrockis** [comment]
Paul - we haven't fully tested and validated this sample yet.  That's why it's still listed as (Coming Soon) in our documentation: https://azure.microsoft.com/en-us/documentation/articles/active-directory-b2c-overview/.  I'm not sure what your error is, but we have another outstanding bug that is preventing this sample from working correctly in all cases.

I wish I could help more but at this time there's not much more I can give you.

**Comment by phatcher** [comment]
Danny, sorry I missed that - I presumed as the Android one was working, the native .NET one was as well.

I'll wait on the release of the sample - would be nice to have a better idea of timelines since it's almost 6 months since the initial drop.

**Comment by baSSiLL** [comment]
I can authenticate successfully using the code based on this sample unless I select **emails** as one of the returned claims for sign-in policy. When **emails** is to be returned, I get exactly the same error.

I guess this may be related to the type of the **emails** attribute which is **StringCollection**. All other attributes are of the **String** type.

**Comment by pengzhisun** [comment]
I've got this error message too..
![image](https://cloud.githubusercontent.com/assets/764635/15067443/00615302-13a3-11e6-8d6b-2cfcb76fd126.png)

My environment:

Windows Server 2012 R2 + Visual Studio 2015 Update2

**Comment by JamieLup** [comment]
getting this error: "The browser based authentication dialog failed to complete. Reason: The protocol is not known and no pluggable protocols have been entered that match." when I attempt to log in with a local user account. But it works when I use an account xx@aaa.onmicrosoft.com. (i.e. an account created on the tenant directly using the portal). This only started happening recently, the only changes i've made are that I have added an additional user attribute and added it to the sign in policy claims.. This issue I'm facing only happens on native application, I have a web application that interacts with the SAME active directory tentant and that works as expected. Please help.

Thanks

**Comment by pengzhisun** [comment]
Thanks @baSSiLL , your solution worked well.

**Comment by dstrockis** [comment]
Thanks folks. We'll be looking into this asap.

**Comment by trietphamvm** [comment]
After I signup a local with B2C, it works fine, but when I edit profile it raises errors:

![image](https://cloud.githubusercontent.com/assets/14874365/17268856/a01c7858-5661-11e6-94e0-eb3bf7ccff40.png)

**Comment by ppthanhtn** [comment]
Shit, did you test your sample yet? A lot of errors happen. You should test carefully and push it here and make a very shit long walkthrough at https://azure.microsoft.com/en-us/documentation/articles/active-directory-b2c-devquickstarts-native-dotnet/. Here are some:

After I signup a local with B2C, it works fine, but when I edit profile it raises errors:

![image](https://cloud.githubusercontent.com/assets/14874365/17268856/a01c7858-5661-11e6-94e0-eb3bf7ccff40.png)

After I sign-up and login with external identity provider. It says "The browser based authentication dialog failed to complete. Reason: The protocol is not known and no pluggable protocols have been entered that match". 

It's silly if I have to sign-up with external identity provider. You should change the lib to implicit sign-up external identity provider and going back the app.

Thanks

**Comment by pottabathini** [comment]
I too have the same error but when I try with **Sign Up and Sign In Policy** it got worked properly and I am able to log in using Google Auth.

**Comment by dstrockis** [comment]
Turns out that the web dialog that MSAL uses to present web pages chokes if the return URL from the authorize endpoint is too long.  This occurs whenever the authorization code returned by B2C is sufficiently large.  I didn't determine the exact limit, but it seems to be around 2K characters. 

In B2C, the size of authorization codes is proportional to the customization of the policy.  So, to workaround this issue, developers can try to remove claims from their policies.  This typically reduces the size of the authorization code.

We are looking at alternatives to solve this problem without workarounds.

**Comment by Bigshooter** [comment]
Any update on this issue?

**Comment by Bigshooter** [comment]
Any update?

**Comment by iloabn** [comment]
Happened to run into the same issue today.
When I have too many claims enabled I get the issue, when I disable some it starts to work.
Is there any other workaround or possibility of documenting this somewhere?
If it's not on the roadmap to be fixed, I think it would be nice to have it mentioned on these sites/samples:

https://docs.microsoft.com/en-us/azure/active-directory-b2c/add-native-application
https://github.com/Azure-Samples/active-directory-b2c-dotnet-uwp

(I could create PRs if I know that the bug isn't prioritized)

**Comment by jmprieur** [comment]
@iloabn : don't hesitate to propose PRs (I understand that this is about updating the README.md of the samples?)

**Comment by nkumars** [comment]
It still appeared now to me, I followed this link https://learn.microsoft.com/en-us/answers/questions/1248915/b2c-login-failed to upgrade **Microsoft.Identity.Client** package to **4.42.0** version and thereafter it all worked like normal
